packages/itwin/grouping-mapping-widget/src/components/Mappings/MappingListItem.tsx

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { Mapping } from "@itwin/insights-client";
+import React, { useCallback, useEffect, useState } from "react";
+import type { BeEvent } from "@itwin/core-bentley";
+import { MappingUIActionGroup } from "./MappingViewActionGroup";
+import { Anchor, ListItem, Text } from "@itwin/itwinui-react";
+import { ExtractionStates } from "./Extraction/ExtractionStatus";
+import { ExtractionStatus } from "./Extraction/ExtractionStatus";
+import { useExtractionStateJobContext } from "../context/ExtractionStateJobContext";
+import { resetMappingExtractionStatus } from "./hooks/useFetchMappingExtractionStatus";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useFetchMappingExtractionStatus } from "./hooks/useFetchMappingExtractionStatus";
+import "./MappingListItem.scss";
+import { useGroupingMappingApiConfig } from "../context/GroupingApiConfigContext";
+
+export interface MappingListItemProps {
+  selected: boolean;
+  onSelectionChange: (mapping: Mapping) => void;
+  mapping: Mapping;
+  jobId: string;
+  jobStartEvent: BeEvent<(mappingId: string) => void>;
+  onClickMappingTitle?: (mapping: Mapping) => void;
+  onClickMappingModify?: (mapping: Mapping) => void;
+  onRefreshMappings: () => Promise<void>;
+  onToggleExtraction: (mapping: Mapping) => Promise<void>;
+  setShowDeleteModal: (mapping?: Mapping) => void;
+  isMappingPageRefreshed?: boolean;
+  setIsMappingPageRefreshed?: (isMappingPageRefreshed: boolean) => void;
+}
+
+export const MappingListItem = (props: MappingListItemProps) => {
+  const [extractionState, setExtractionState] = useState<ExtractionStates | undefined>(ExtractionStates.None);
+  const { mappingIdJobInfo, setMappingIdJobInfo } = useExtractionStateJobContext();
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const [isJobStarted, setIsJobStarted] = useState<boolean>(false);
+  const [isListItemMounted, setIsListItemMounted] = useState<boolean>(false);
+  const { statusQuery } = useFetchMappingExtractionStatus({isMounted: isListItemMounted, ...groupingMappingApiConfig, mapping: props.mapping, enabled: isJobStarted});
+  const queryClient = useQueryClient();
+
+  const onClickTile = () => {
+    props.onSelectionChange(props.mapping);
+  };
+
+  useEffect(() => {
+    // Only apply to all mappings when the page is reloaded, not the list component
+    // or when users modify mappings
+    if(props.isMappingPageRefreshed && props.setIsMappingPageRefreshed){
+      setIsListItemMounted(true);
+      setIsJobStarted(true);
+      props.setIsMappingPageRefreshed(false);
+    }
+  }, [props]);
```

Comment:
Is this on purpose or does it need to be this broad? Couldn't it just be `[props.isMappingPageRefreshed, props.setIsMappingPageRefreshed]`

Reply:
For some reason, eslint errors on my end. Should I just include the exhaustive deps comment?
![image](https://github.com/iTwin/viewer-components-react/assets/56598021/efdd2cd3-a6f6-4ef0-a187-ace03b375728)

Reply:
Hmm. Just get rid of props. Deconstruct it in the params.

Reply:
I have updated the params.